arch/x86_64/CMakeLists.txt

```diff
@@ -11,4 +11,7 @@ add_subdirectory(core)
 
 zephyr_compile_options(${X86_64_BASE_CFLAGS} -mx32)
 
-zephyr_link_libraries(-mx32)
+zephyr_link_libraries(
+    -mx32
+    -Wl,-melf32_x86_64
```

Comment:
For portability, "-Wl" should be replaced with ${LINKERFLAGPREFIX} (after verifying that LINKERFLAGPREFIX is intialized before use).

Reply:
Replaced `-Wl` with `${LINKERFLAGPREFIX}`. The flag should have been defined as these run during the last few steps of the build process (or it would have error-ed out earlier). I don't see anywhere else actually checking for it being defined though.